src/fns.ts

```diff
@@ -50,23 +64,100 @@ export const rateLimit = <T = unknown>(limit: number) => {
    * Wait for all promises to resolve
    */
   const finish = async () => {
-    debug('finish')
+    rl('finish')
     await Promise.all(set)
   }
   return { add, finish }
 }
 
 /**
+ * Limit throughput by sleeping until the rate (items/sec)
+ * is less than or equal to `maxItemsPerSec`.
  *
+ * Example:
+ * ```typescript
+ * const limiter = throughputLimiter(1000)
+ *
+ * for(const batch of batches) {
+ *   // Will wait until the rate is <= `maxItemsPerSec`
+ *   await limiter.throttle(batch.length)
+ *   console.log('Items/sec %d', limiter.getCurrentRate())
+ * }
+ * ```
+ */
+export const throughputLimiter = (
+  maxItemsPerSec: number,
```

Comment:
Do you think it might make sense, at some point, to add functionality to limit throughput by bytes/sec instead of items/sec? If so, it might be worth making `maxItemsPerSec` one of the `options`, instead of a separate argument. That way, if we ever wanted to add a `maxBytesPerSec` option, we could do it without making a breaking change to the function signature.

Reply:
Never mind. I just saw below how you can use `throughputLimiter` with a number of bytes instead of a number of items. Mind blown! 🤯

Reply:
`throughputLimiter` deals in `unitsPerSec`, really. That may be a better name here. `batchQueue` uses `throughputLimiter` for items/sec and bytes/sec.

Reply:
Also agree that it would make sense to call the parameter something different because of the overlap with the batchQueue option parameter, I think `unitsPerSec` sounds good

Reply:
Changed.